_docs/durability.md

```diff
@@ -4,4 +4,138 @@ title: Durability
+                                 ;; in this example.
+                                 {:base-rulebase deserialized-rulebase})))
+
+{% endhighlight %}
+
+This example is the same as before, except the session rulebase and working memory are serialized separately.  This is commonly a very useful feature.  It is often the case that a "base" session is created that constructs a rulebase (Rete-based rule/query graph).  This base session is then used for mutually exclusive working memory datasets e.g. per-person domain data processed against the same set of rules.  For large sets of rules/queries, the rulebase can take non-trivial amounts of space and time to serialize and deserialize.  If the rulebase can be shared among different working memory states, Clara durability allows it to be stored separately.  When deserializing the working memory state to recreate a deserialized session, the deserialized rulebase just has to be provided via the `:base-rulebase` option passed to `d/deserialize-session-state` above.
+
+### Implementing the d/IWorkingMemorySerializer protocol ###
+
+As stated previously, Clara currently provides a default implementation of the `d/ISessionSerializer` that uses [Fressian](https://github.com/clojure/data.fressian), but it does not currently provide a default implementation for the `d/IWorkingMemorySerializer`
+
+The reason for this is that the `d/ISessionSerializer` is responsible for serializing and deserializing the actual rulebase Rete-based graph representation.  This is really an implementation detail of Clara.  Any implementor of this protocol is going to become necessarily coupled to quite a few internal details of this graph structure.  Also, this structure is fully within the domain of Clara rules itself.  This makes it reasonable, and perhaps necessary, for Clara to maintain an efficient implementation (and in the future possibly more) of this protocol.  Most consumers are expected to just use the provided implementation(s) as-is.
+
+The `d/IWorkingMemorySerializer` protocol is different.  It is responsible for serializing and deserializing the state of the user-defined facts from a domain unknown to Clara rules itself.  If the domain of facts happens to be Clojure datatypes that are supported by Fressian and Clara's `clara.rules.durability.fressian` custom Fressian type extensions, it may be reasonable to offer this as a default implementation from Clara that could optionally be used by consumers.  This may happen the future.
+
+However, in general, it is difficult to provide a suitable serialization protocol for user-defined facts, since they can be many different datatypes and structures.  The facts could be Java POJOs, that maybe should be serialized via [Java serialization](https://docs.oracle.com/javase/8/docs/technotes/guides/serialization/); the facts could be [Avro](https://avro.apache.org/) that have their own specific serialization format; etc.
+
```

Comment:
I still don't agree.  I'm providing a few concrete examples of things that can happen.  It just makes it clear to provide a concrete example to support an abstract phrase.

This really has no domain-specifics involved at all.  Just because you may or may not use Java/Avro at a certain company.  These are very general situations.

In a pure Clojure-world, most things would just be find with Fressian or a handful of others.

However, in any Java interop applications, Java serialization is a very likely one that would come up.  Also, other serialization specific "POJO"s like Avro/Thrift records etc.

I do not see this as a blog post at all.  It is just helpful to provide real-world examples to support abstractions.

Also, it actually would be beneficial for Clara to provide some sort of hybrid d/IWorkingMemorySerializer protocol in the future, with perhaps some "optional" dependency components too it.  I'd think Java serialization would be something to support in this.  As well as possibly something like Avro.  These are generally useful and implementing the d/IWorkingMemorySerializer in an efficient and practical way isn't really very trivial for consumers.

I think it is one of the weakest points in the provide Clara durability impl right now.